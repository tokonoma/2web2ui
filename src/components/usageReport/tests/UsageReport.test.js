import React from 'react';
import { shallow } from 'enzyme';
import {
  UsageReport,
  OGDisplayNumber,
  HibanaDisplayNumber,
  HibanaProgressLabel,
  OGProgressLabel,
} from '../UsageReport';

describe('UsageReport Component', () => {
  let getAccount;
  let props;

  beforeEach(() => {
    getAccount = jest.fn();

    props = {
      subscription: {
        plan_volume: 10000,
      },
      usage: {
        month: {
          used: 1000,
          limit: 50000,
          start: '2017-08-01T08:00:00.000Z',
          end: '2017-08-31T08:00:00.000Z',
        },
        day: {
          used: 1000,
          limit: 2000,
          start: '2017-08-30T00:00:00.000Z',
        },
      },
      getAccount,
    };
  });

  it('should render null without props', () => {
    const props = { getAccount };
    const wrapper = shallow(<UsageReport {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render null when usage is zero', () => {
    const props = {
      getAccount: getAccount,
      subscription: {
        plan_volume: 10000,
      },
      usage: {
        month: {
          used: 0,
          limit: 50000,
          start: '2017-08-01T08:00:00.000Z',
          end: '2017-08-31T08:00:00.000Z',
        },
        day: {
          used: 0,
          limit: 2000,
          start: '2017-08-30T00:00:00.000Z',
        },
      },
    };
    const wrapper = shallow(<UsageReport {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with regular usage', () => {
    const wrapper = shallow(<UsageReport {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(shallow(<OGDisplayNumber />)).toMatchSnapshot();
    expect(shallow(<HibanaDisplayNumber />)).toMatchSnapshot();
    expect(shallow(<OGProgressLabel />)).toMatchSnapshot();
    expect(shallow(<HibanaProgressLabel />)).toMatchSnapshot();
  });

  it('should get the account including usage', () => {
    shallow(<UsageReport {...props} />);
    expect(props.getAccount).toHaveBeenCalledWith({ include: 'usage' });
  });

  it('should render with overages', () => {
    props.usage.month.used = props.subscription.plan_volume + 1000;
    const wrapper = shallow(<UsageReport {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render limits for unrestricted accounts', () => {
    const usage = {
      day: {
        used: 1000,
        limit: null,
        start: '2017-08-30T00:00:00.000Z',
      },
      month: {
        used: 1000,
        limit: null,
        start: '2017-08-01T08:00:00.000Z',
        end: '2017-08-31T08:00:00.000Z',
      },
    };
    const wrapper = shallow(<UsageReport {...props} usage={usage} />);

    expect(wrapper.find('ProgressBar')).not.toExist();
    expect(wrapper.find('DisplayNumber[label="Daily limit"]')).not.toExist();
    expect(wrapper.find('DisplayNumber[label="Monthly limit"]')).not.toExist();
  });
});
