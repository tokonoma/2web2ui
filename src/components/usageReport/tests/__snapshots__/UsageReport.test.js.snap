// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsageReport Component should render null when usage is zero 1`] = `""`;

exports[`UsageReport Component should render null without props 1`] = `
<PanelLoading
  accent={false}
  minHeight="400px"
/>
`;

exports[`UsageReport Component should render with overages 1`] = `
<Panel
  actions={
    Array [
      Object {
        "color": "orange",
        "content": "What counts towards usage?",
        "external": true,
        "to": "https://www.sparkpost.com/docs/getting-started/what-counts-daily-monthly-usage",
      },
    ]
  }
  title="Your Usage Report"
>
  <Panel.Section>
    <ProgressLabel
      secondaryTitle="Since Aug 29 2017, 8:00pm"
      title="Today"
    />
    <ProgressBar
      completed={50}
      my="300"
    />
    <DisplayNumber
      content="1,000"
      label="Used"
      primary={true}
    />
    <DisplayNumber
      content="2,000"
      label="Daily Limit"
    />
    <SendMoreCTA
      hasSendingLimits={true}
    />
  </Panel.Section>
  <Panel.Section>
    <ProgressLabel
      secondaryTitle="Billing cycle: Aug 1 2017 - Aug 31 2017"
      title="This Month"
    />
    <ProgressBar
      completed={22}
      my="300"
    />
    <DisplayNumber
      content="11,000"
      label="Used"
      primary={true}
    />
    <DisplayNumber
      content="10,000"
      label="Included"
    />
    <DisplayNumber
      content="1,000"
      label="Extra Emails Used"
    />
    <DisplayNumber
      content="50,000"
      label="Monthly limit"
    />
  </Panel.Section>
</Panel>
`;

exports[`UsageReport Component should render with regular usage 1`] = `
<Panel
  actions={
    Array [
      Object {
        "color": "orange",
        "content": "What counts towards usage?",
        "external": true,
        "to": "https://www.sparkpost.com/docs/getting-started/what-counts-daily-monthly-usage",
      },
    ]
  }
  title="Your Usage Report"
>
  <Panel.Section>
    <ProgressLabel
      secondaryTitle="Since Aug 29 2017, 8:00pm"
      title="Today"
    />
    <ProgressBar
      completed={50}
      my="300"
    />
    <DisplayNumber
      content="1,000"
      label="Used"
      primary={true}
    />
    <DisplayNumber
      content="2,000"
      label="Daily Limit"
    />
    <SendMoreCTA
      hasSendingLimits={true}
    />
  </Panel.Section>
  <Panel.Section>
    <ProgressLabel
      secondaryTitle="Billing cycle: Aug 1 2017 - Aug 31 2017"
      title="This Month"
    />
    <ProgressBar
      completed={2}
      my="300"
    />
    <DisplayNumber
      content="1,000"
      label="Used"
      primary={true}
    />
    <DisplayNumber
      content="10,000"
      label="Included"
    />
    <DisplayNumber
      content="50,000"
      label="Monthly limit"
    />
  </Panel.Section>
</Panel>
`;

exports[`UsageReport Component should render with regular usage 2`] = `
<div
  className="Display"
>
  <h5
    className="Content"
  />
  <h6
    className="Label"
  />
</div>
`;

exports[`UsageReport Component should render with regular usage 3`] = `
<Box
  display="inline-block"
  mr="500"
>
  <Text
    fontSize="400"
    fontWeight="medium"
  />
  <Text
    fontSize="200"
    fontWeight="light"
  />
</Box>
`;

exports[`UsageReport Component should render with regular usage 4`] = `
<div>
  <h5
    className="ProgressTitle"
  />
  <h6
    className="ProgressSecondary"
  />
</div>
`;

exports[`UsageReport Component should render with regular usage 5`] = `
<Box
  mb="200"
>
  <Inline>
    <Text
      fontSize="200"
      fontWeight="semibold"
    />
    <Text
      color="gray.600"
      fontSize="200"
      fontWeight="light"
    />
  </Inline>
</Box>
`;
